lib/beetle/message.rb

```diff
@@ -193,6 +200,16 @@ def set_delay!
       @store.set(msg_id, :delay, now + next_delay(attempts))
     end
 
+    # whether we should wait before running the handler
+    def deferred?
+      (t = @store.get(msg_id, :defer)) && t.to_i > now
+    end
+
+    # store delay value in the deduplication store
+    def set_defer!
+      @store.setnx(msg_id, :defer, now + defer)
+    end
```

Comment:
set if not exists, ensures the first handler run sets it then delays and the next runs just check for still deferred?